scipy/spatial/transform/rotation.py

```diff
@@ -1461,6 +1518,103 @@ def magnitude(self):
         else:
             return angles
 
+    def reduce(self, other):
+        """Use this rotation to reduce the other.
+
+        Parameters
+        ----------
+        other : `Rotation` instance
+            Object containing the rotations to be reduced.
+
+        Returns
+        -------
+        reduced : `Rotation` instance
+            Object containing reduced rotations.
+
+        Examples
+        --------
+        >>> from scipy.spatial.transform import Rotation as R
+        >>> r = R.from_quat([0.5, 0.5, 0.5, 0.5])
+        >>> R.create_group("O").reduce(r).as_quat()
+        array([0., 0., 0., 1.])
+
+        >>> R.create_group("C2").reduce(r).as_quat()
+        array([0.5, 0.5, 0.5, 0.5])
+        """
+
+        quat = self.as_quat().reshape((len(self), 4))
+        qother = other.as_quat().reshape((len(other), 4))
+
+        cosines = np.outer(quat[:, 3], qother[:, 3])
+        cosines -= np.outer(quat[:, 0], qother[:, 0])
+        cosines -= np.outer(quat[:, 1], qother[:, 1])
+        cosines -= np.outer(quat[:, 2], qother[:, 2])
+        cosines = np.minimum(1, np.abs(cosines))
+
+        return self * other[np.argmax(cosines, axis=1)]
+
+    @classmethod
+    def create_group(cls, group, axis='Z'):
+        """Get a 3D rotation group.
+
+        Parameters
+        ----------
+        group: string
+            The name of the group.  Must be one of 'I', 'O', 'T', 'Cn', 'Dn',
+            where `n` is a positive integer.
+        axis: integer
+            The cyclic rotation axis Must be one of 'X', 'Y', 'Z'.
+            Default is 'Z'.  Ignored for groups 'I', 'O', and 'T'.
+
+        Returns
+        -------
+        rotation : `Rotation` instance
+            Object containing the elements of the rotation group.
+
+        Raises
+        ------
+        ValueError
+            If the group order (`n`) is not a positive integer.
+
+        Notes
+        -----
+        This method generates rotation groups only.  The full 3-dimensional
+        groups also contain reflections.  See e.g. [Conway]_ for a good
+        description of the rotation groups and full 3-dimensional groups.
+
+        References
+        ----------
+        .. [Conway] John H Conway and Derek A Smith, On quaternions and
+                    octonions, 2003, AK Peters/CRC Press, ISBN 978-1568811345
+        """
+
+        if type(group) != str:
```

Comment:
Basically you can move this code into a separate file, pass `cls` to the function if you want.

Reply:
Done.